include/swift/Sema/CSDiag.h

```diff
@@ -0,0 +1,32 @@
+//===--- CSDiag.h - Constraint-based Type Checking ----*- C++ -*-===//
```

Comment:
This should be in lib/Sema/CSDiag.h, just like ConstraintSystem.h etc. It's not used by other subsystems that depend on Sema, only Sema internally. Only the former go into include/swift/Sema.